oauthproxy.go

```diff
@@ -279,6 +285,7 @@ func (p *OAuthProxy) buildServeMux(proxyPrefix string) {
 	r := mux.NewRouter().UseEncodedPath()
 	// Everything served by the router must go through the preAuthChain first.
 	r.Use(p.preAuthChain.Then)
+	r.Use(p.sessionChain.Then)
```

Comment:
This means we will go through the session chain twice now for the paths that need it, why do the paths that previously didn't need this chain now need it?